primo/opt_model/result_parser.py

```diff
@@ -0,0 +1,776 @@
+            name
+            for name in names_attributes
+            if any([name2 in name for name2 in self.accessibility_attr])
+        ]
+        if len(names_attributes_accessibility) == 0:
+            return None
+        total_weight = sum(
+            int(name.split("_0_")[-1]) for name in names_attributes_accessibility
+        )
+        return total_weight, sum(
+            getattr(self, attribute) for attribute in names_attributes_accessibility
+        )
+
+    def _add_distance_to_centroid_col(self):
+        """
+        Adds the distance to centroid column to the well data
+        """
+        # I am assuming here that apply returns the data in the corresponding order
+        self.well_data.add_new_column_ordered(
+            ["dist_centroid", "Distance to Centroid [miles]"],
+            self.well_data.data.apply(
+                lambda row: get_distance(
+                    (row[self._col_names.latitude], row[self._col_names.longitude]),
+                    self.centroid,
+                ),
+                axis=1,
+            ).values,
+        )
+
+    def get_max_val_col(self, col_name) -> float:
+        """
+        Returns the maximum value of a column
+
+        Parameters
+        ----------
+        col_name : str
+            column name
+
+        Returns
+        -------
+        float
+            the maximum value of the column
+
+        """
+        return self.well_data[col_name].max()
+
+    def update_efficiency_score(self, value: Union[int, float]):
+        """
+        Updates the efficiency score of a project
+
+        Parameters
+        ----------
+        value : Union[int, float]
+            The value to update the efficiency score with
+        """
+        if len(self.well_data) == 1:
+            return
+        self.efficiency_score += value
+
+    def get_well_info_dataframe(self):
+        """
+        Returns the data frame to display in the notebook
+        """
+        return self.well_data[self._essential_cols]
+
+
+class OptimalCampaign:
+    """
+    Represents an optimal campaign that consists of multiple projects.
+    """
+
+    def __init__(self, wd: WellData, clusters_dict: dict, plugging_cost: dict):
+        """
+        Represents an optimal campaign that consists of multiple projects.
+
+        Parameters
+        ----------
+        wd : WellData
+            WellData object
+
+        clusters_dict : dict
+            A dictionary where keys are cluster numbers and values
+            are list of wells for each cluster.
+
+        plugging_cost : dict
+            A dictionary where keys are cluster numbers and values
+            are plugging cost for that cluster
+        """
+        # for now include a pointer to well data, so that I have column names
+        self.wd = wd
+        self.projects = {}
+
+        index = 1
+        for cluster, wells in clusters_dict.items():
+            self.projects[index] = OptimalProject(

```

Comment:
Any specific reason to use a numbered index for projects instead of project_id as key directly? Would help you not having to separately calculate/store the map later

Reply:
@radhakrishnatg @DevKakkad32 I received an initial class file and this is an artifact of that file. I think the reason is that we would like to display "Project 1, Project 2, etc.", but the way it is currently implemented (and was in the previous notebook), we actually use the cluster number to id the projects (e.g., Project 5, Project 17, etc.) I think it is best to simply go with the index, and id the projects with the index so there is no mapping back and forth, but there may be some background that I am missing.

Reply:
That is correct. Earlier, cluster number (which is arbitrary) was used to label projects. The user would not be aware of the clusters, so we can just index them as 1, 2, 3,... and display them as it is. We don't have to map them to clusters.

Reply:
Addressed.